src/components/MarketChartData/MarketChartData.js

```diff
@@ -60,7 +60,7 @@ export default class MarketChartData extends React.Component {
   };
 
   setCurrency = () => {
-    let currency = this.props.currency;
+    const currency = this.props.currency;
     this.setState({ currency });
     this.getChartData(this.state.currency);
```

Comment:
you know here you are fetching data with the previous currency, right?

Reply:
now I do :0
and i see the redundancy. took me a while to figure out passing the currency prop to the other components but should be all sorted now